src/compiler/checker.ts

```diff
@@ -18164,6 +18164,10 @@ namespace ts {
             return false;
         }
 
+        function containsUndefinedType(type: Type) {
+            return !!((type.flags & TypeFlags.Union ? (type as UnionType).types[0] : type).flags & TypeFlags.Undefined);

```

Comment:
We have to be careful with this - nothing with typeflags other than `Unknown` and `Any` can now precede the group of `Undefined`-ish types starting with `undefinedType` until `missingType`.

Reply:
Agreed, but we already do this in a few places in the interest of performance.